hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -395,24 +390,17 @@ private Map<String, HoodieRecord<HoodieMetadataPayload>> fetchBaseFileRecordsByK
       result = toStream(records)
           .map(record -> {
             GenericRecord data = (GenericRecord) record.getData();
-            return Pair.of(
-                (String) (data).get(HoodieMetadataPayload.KEY_FIELD_NAME),
-                composeRecord(data, partitionName));
+            String recordKey = (String) data.get(HoodieMetadataPayload.KEY_FIELD_NAME);
+            return Pair.of(recordKey, composeRecord(data, recordKey, partitionName));
           })
           .collect(Collectors.toMap(Pair::getKey, Pair::getValue));
     }
     return result;
   }
 
-  private HoodieRecord<HoodieMetadataPayload> composeRecord(GenericRecord avroRecord, String partitionName) {
-    if (metadataTableConfig.populateMetaFields()) {
-      return SpillableMapUtils.convertToHoodieRecordPayload(avroRecord,
-          metadataTableConfig.getPayloadClass(), metadataTableConfig.getPreCombineField(), false);
-    }
-    return SpillableMapUtils.convertToHoodieRecordPayload(avroRecord,
-        metadataTableConfig.getPayloadClass(), metadataTableConfig.getPreCombineField(),
-        Pair.of(metadataTableConfig.getRecordKeyFieldProp(), metadataTableConfig.getPartitionFieldProp()),
-        false, Option.of(partitionName), Option.empty());
+  private HoodieRecord<HoodieMetadataPayload> composeRecord(GenericRecord avroRecord, String recordKey, String partitionName) {
```

Comment:
can we left some comments here the MDT `populateMetaFields` is hardcoded as false.

Reply:
I'm not sure what you mean, this is preserving the old behavior

Reply:
Or do you mean in the methods calling this where the record key is lookup from the key field instead of the hoodie metadata field, we want a comment saying that the metadata fields are not present?

Reply:
> we want a comment saying that the metadata fields are not present?

yes.